lib/_stream_writable.js

```diff
@@ -294,34 +294,32 @@ function validChunk(stream, state, chunk, cb) {
 
 Writable.prototype.write = function(chunk, encoding, cb) {
   const state = this._writableState;
-  var ret = false;
-  const isBuf = !state.objectMode && Stream._isUint8Array(chunk);
-
-  // Do not use Object.getPrototypeOf as it is slower since V8 7.3.
-  if (isBuf && !(chunk instanceof Buffer)) {
-    chunk = Stream._uint8ArrayToBuffer(chunk);
-  }
 
   if (typeof encoding === 'function') {
     cb = encoding;
     encoding = null;
   }
 
-  if (isBuf)
+  if (!state.objectMode && Stream._isUint8Array(chunk)) {
+    if (!(chunk instanceof Buffer)) {
+      chunk = Stream._uint8ArrayToBuffer(chunk);
+    }
```

Comment:
@addaleax: I don't quite understand the thing with `_isUint8Array`. 

Shouldn't this:

```js
const isBuf = !state.objectMode && Stream._isUint8Array(chunk);
```

be:


```js
const isBuf = !state.objectMode && (chunk instanceof Buffer || Stream._isUint8Array(chunk));
```

?